### ISSUE TITLE
[Open GL] Arc The Lad III Disc Switch is not recognized.

### ISSUE THREAD

**Issue by OCRBonk** [libretro/beetle-psx-libretro#363]
Sort of a repost of my post on the forums.

Hit the disc switch point and NOTHING will cause the game to recognize disc changes.
Using a playlist, manually switching the image files.

I hit eject, return to the game, let it set, change the disc to disc 2, change back to the game, let it sit, then go back and close the disc tray. And nothing. I thought it might be my images, both discs load fine individually. But neither will allow a disc swap. I made a new image of disc 2 from my disc, made sure the SHA-1 hash matched redump. That didn’t work. Tried an image from the redump set online. That didn’t work.

Tried both software and hardware. I tried Mednafen standalone with a playlist file. Allows to boot disc 1 or 2, swapping doesn’t work. Tried PCSX-R, that didn’t work either.

On real hardware, the game reacts immediately to the disc tray opening. It will say “Please wait” while the tray is open.

That doesn’t happen at all with Mednafen, the beetle core (SW/HW) or PCSX-R PXGP
However it works in PSX 1.13. The eject CD function actually works here and allows the game to switch discs properly.

Memory Card saves 
http://www.mediafire.com/file/w99nvrupxy9qh94/CDROM_2.1.mcr
http://www.mediafire.com/file/cb9a7th8u4q51sm/CDROM_2.srm
If you wish to test, it is the first 2 save files at around level 48 in Cariote Cave. One is before the point of no return and one is at the point of no return.

Windows 7 x64
i7 950
GTX 980 388.59 drivers
Retroarch build 1.7.2 (April 13,2018)
Using the latest downloaded cores.

**Comment by saftle** [comment]
Did you follow the steps detailed here?:

https://github.com/libretro/beetle-psx-libretro/issues/252#issuecomment-341304955

**Comment by OCRBonk** [comment]
Yeah, I know how to switch discs. But the game doesn't recognize at any point that the tray has been opened. I've tried it every which way. As soon as the tray is opened on real hardware, the game immediately changes the message from "Please insert disc 2/1" to "Please wait". 

No matter how you do it, this never happens in Retroarch (Or on Mednafen standalone). You hit open tray and go back to the game, and nothing happens. I tried even turning off the option to pause emulation when in the menu. That did nothing as well.
PCSX-R's disc switch does not work either, I can't speak for EPSXE or XEBRA. But PSX was the only emulator that it worked with. (And I discovered that after I performed the switch on real hardware instead. Dex drived my save back and continued playing.)

The provided memory card files have save files at the switch point. People are free to try it themselves and verify it.

**Comment by Metasynic** [comment]
This has now been fixed on [mednafen standalone](https://forum.fobby.net/index.php?t=rview&goto=5736&th=1752#msg_5736) so hopefully the change will end up here soon.

**Comment by klepp0906** [comment]
just for an update, since im having a hard time following this since its been referenced in several places...

is arc the lad III disc switching working now or?

I dont want to sink a ton of hours into something thats going to run me into a brick wall, so figured id check as this was next on my hit list ;p

**Comment by rz5** [comment]
@klepp0906 - It's not working, no. Until issue #487 is dealt with, it won't work.

**Comment by OCRBonk** [comment]
You should be able to make the switch in another emu and then reimport the save if you have to. Like standalone, or psx.  Or if you have the game and a dex drive. Complete the switch on real he like I did.

**Comment by klepp0906** [comment]
ah, i didnt realize saves were cross compatible.  If it comes down to it, at least i know thats an option now.  thanks :)

**Comment by OCRBonk** [comment]
If you have issues you may be able to convert using memcardrex as well.

**Comment by abt462** [comment]
Hello,
I noticed yesterday that the issue still exists in the current version of the core (0.9.44.1).

From what I understand, there were plans to apply changes from the new (at the time) version of Mednafen, but they were reverted as another version was released and I'm not sure how/where it ended up.

I.e. I don't know if the fix for this specific bug was implemented and then reverted, if it was left aside for some reason and then forgotten about...
